Sources/AppcuesKit/Presentation/DeepLinkHandler.swift

```diff
@@ -131,6 +131,27 @@ internal class DeepLinkHandler: DeepLinkHandling {
         }
     }
 
+    private func previewCompletion(_ result: Result<Void, Error>) {
+        guard case let .failure(error) = result else {
+            return
+        }
+
+        let message: String
+        switch error {
+        case let ExperienceRendererError.renderDeferred(context, experience):
+            message = "Please navigate to the screen with \(context.description) to preview \(experience.name)."
+        case NetworkingError.nonSuccessfulStatusCode(404):
+            message = "Experience not found."
+        case is NetworkingError:
+            message = "Error loading mobile flow preview."
+        default:
+            message = "Mobile flow preview failed."
+        }
+
+        let toast = DebugToast(message: .custom(text: message), style: .failure)
+        container?.resolve(UIDebugging.self).showToast(toast)
+    }
```

Comment:
nice, awesome that we already had the completion plumbing in place! this fit in nicely. So a failed tooltip for a missing target would just be `Mobile flow preview failed` right? I think thats good, nicely delineates the possible reasons something might not show.

@andretortolano I think we'll have to change `ExperienceRenderer.show()` to return a `ResultOf` instead of a `Boolean` to do something like this and get the actual reasons for failure.

Reply:
Correct. I can add a case for `ExperienceStateMachine.ExperienceError` and show the experience error if we want.

(I can also add support for >2 lines)

![Simulator Screenshot - iPhone 14 Pro - 2023-08-21 at 13 36 13](https://github.com/appcues/appcues-ios-sdk/assets/845681/62a83f87-84c9-4bea-ac45-05740a94bc58)